Fix issues after running static analysis: part 4

amiclaus opened the pull request:
Fourth set of fixed issues after running cppcheck tool.

Signed-off-by: Antoniu Miclaus antoniu.miclaus@analog.com

dbogdan commented on drivers/adf4350/adf4350.c:
```diff
@@ -248,7 +248,7 @@ int64_t adf4350_set_freq(adf4350_dev *dev,
 		ADF4350_REG2_LDP_6ns | ADF4350_REG2_LDF_INT_N |
 		ADF4350_REG2_CHARGE_PUMP_CURR_uA(5000) |
-		ADF4350_REG2_MUXOUT(0x7) | ADF4350_REG2_NOISE_MODE(0x9)));
+		ADF4350_REG2_MUXOUT(0x7) | ADF4350_REG2_NOISE_MODE(0b101)));
```
ADF4350_REG2_NOISE_MODE has a length of 2 bits, so 0b101 value is incorrect. I would also specify this in its definition (e.g., (((x) & 0x3) << 29)).
In this place, ADF4350_REG2_NOISE_MODE should be set 0x3 to enable the low spur mode.

amiclaus commented:
Changelog: v1->v2:
* added "Signed-off by:" to each commit.
* added description to each commit.
* modified macro definition for ADF4350_REG2_NOISE_MODE and set value to enable low spur mode.

commodo commented on adum7701_fmc/clkgen_core.c:
```diff
@@ -425,5 +425,7 @@ int32_t clkgen_setup(clkgen_device **clkgen_dev,
 		status--;
 	}
 
+	free (adc_clkgen);
```
I think in this case you could probably change the type of the `adc_clkgen` variable;
to `clkgen_device adc_clkgen;`

that way it's automatically allocated on & free'd from the stack

commodo replied:
this can be done here, also because you are allocating a fixed size memory block, so malloc() doesn't add much value

commodo commented on common_drivers/platform_drivers/platform_drivers.c:
```diff
@@ -196,6 +196,7 @@ int32_t spi_init(spi_desc **desc,
 		break;
 #endif
 	default:
+		free(dev);
```
this is good;
but i think you may also need to add the` free()` at lines 210-211 & 214-215 below

commodo commented on common_drivers/platform_drivers/platform_drivers.c:
```diff
@@ -315,6 +316,7 @@ int32_t gpio_get(gpio_desc **desc,
 	if (gpio_number < 32) {
+		free(dev);
 		return FAILURE;
 	}
```
in this case, if you move this code block before the malloc() you don't need to do the free()

commodo commented on drivers/linux_platform/platform_drivers.c:
```diff
@@ -192,20 +193,23 @@ int32_t spi_init(spi_desc **desc, const spi_init_param *param)
 			&param->mode);
 	if (ret == -1) {
 		printf("%s: Can't set mode\n\r", __func__);
+		free(descriptor);
```
i think here you may also need to close the `fd` before free-ing the descriptor
so,
```
close(descriptor->fd);
free(descriptor);
```
similarly, this should be done at lines 212 & 300

amiclaus commented:
Changelog v2->v3:
 * remove unnecessary memory deallocation by moving 'if' statement.
 * close 'fd 'before the memory is freed

commodo commented on adum7701_fmc/clkgen_core.c:
```diff
@@ -404,14 +404,10 @@ uint32_t clkgen_get_rate(clkgen_device *dev,
-        if (!adc_clkgen)
-                return -1;
-
 	adc_clkgen->base_addr = init_params.base_addr;
```
if you change the type here, then this also changes to `adc_clkgen.base_addr = init_params.base_addr;`
and you need to call it with `clkgen_set_rate(&adc_clkgen,` and `clkgen_read(&adc_clkgen,`

commodo commented:
rest looks good

amiclaus commented:
Changelog v3->v4:
 * fix the way `adc_clkgen` structure is accessed after changing the variable type.
